base/uk.ac.stfc.isis.ibex.scriptgenerator/src/uk/ac/stfc/isis/ibex/scriptgenerator/generation/ParametersConverter.java

```diff
@@ -0,0 +1,100 @@
+package uk.ac.stfc.isis.ibex.scriptgenerator.generation;
+
+import java.util.List;
+import java.util.Map;
+import com.google.gson.Gson;
+
+/**
+ * Holds the content for converting user data to json format and also used for
+ * holding content from json file
+ * 
+ * @author mjq34833
+ *
+ */
+public class ParametersConverter {
+
+	String version;
+	String scriptGeneratorVersion;
+	String date;
+	String time;
+	String configurationFilePath;
+	String configurationFileGitHash;
+	String geniePythonVersion;
+	String configurationContent;
+	/**
+	 * Parameter values mapped to its actions
+	 */
+	List<Map<String, String>> actions;
+	
+	/**
+	 * Constructor to initialise member variables required for JSON to Java class and Java Class
+	 * to JSON conversion
+	 * @param version
+	 * @param actions
+	 * @param content
+	 * @param scriptGeneratorVersion
+	 * @param date
+	 * @param time
+	 * @param configurationPath
+	 * @param configuraitonFileGitHash
+	 * @param geniePythonVersion
+	 */
+	ParametersConverter(String version, List<Map<String, String>> actions, String content,
+			String scriptGeneratorVersion, String date, String time, String configurationPath,
+			String configuraitonFileGitHash, String geniePythonVersion) {
```

Comment:
Can we not use configuration but scriptDefinition

Reply:
Done.